db/active_replicator_pull.go

```diff
@@ -45,8 +45,7 @@ func (apr *ActivePullReplicator) Start(ctx context.Context) error {
 	}
 
 	apr.setState(ReplicationStateStarting)
-	// intentionally reset the context from having db information on it?
-	logCtx := base.LogContextWith(ctx, &base.LogContext{CorrelationID: apr.config.ID + "-" + string(ActiveReplicatorTypePull)})
+	logCtx := base.CorrelationIDLogCtx(ctx, apr.config.ID+"-"+string(ActiveReplicatorTypePull))
```

Comment:
Did you sanity check that this wasn't intentional to do something like removing cancellation context?

Reply:
`LogContextWith` keeps the context and just resets the entirety of `base.LogContext`.
To drop a cancelation, you have to do `base.NonCancellableContext` or go added in 1.21 `context.Context.WithoutCancel`